File: bff/Season42.Bff/MovieMatch.cs
namespace Season42.Bff;

/// <summary>
/// A movie TMDB matched a search on: its id and its title, and nothing else. This is the whole
/// of what a search answers with — an overview and a poster belong to the details of one match,
/// which is a separate ask.
/// </summary>
/// <remarks>
/// The id is TMDB's, and it is what the app asks the next question with. It is never stored:
/// the Library holds what the user copied, not a link back to someone else's record.
/// </remarks>
/// <remarks>
/// A movie is titled where a series is named, and each is spelled the way TMDB spells it, so
/// that nothing between here and TMDB has to remember a renaming of its own.
/// </remarks>
public sealed record MovieMatch(int Id, string Title);

File: app/Season42/Library/Search/MovieMatch.swift
import Foundation

/// A movie a search matched, as the BFF answers with it: an id and a title, and nothing else.
/// What a search lists, so that the user can tell which of several similar titles is theirs.
///
/// Someone else's data, never stored. The id is TMDB's and is only ever the thing the next
/// question is asked with — the Movie Details the user opens, and nothing else. Nothing in the
/// Library ever holds one (ADR-0002).
///
/// `Hashable` because it is what a row is pushed with: the match itself is the value the
/// navigation destination is keyed on, so nothing has to be looked up again on the way there.
struct MovieMatch: Decodable, Hashable, Identifiable, Sendable {
    /// TMDB's id, unique among matches, which is what makes it the row's identity.
    let id: Int

    /// A movie is titled where a series is named, and each is spelled the way the BFF spells it.
    let title: String
}
